Update jsf-2.3 to use MyFaces 2.3.10

volosied opened the pull request:
fixes #21501


Last update was on May 4, 2021. 

Overlay Notes: 
 MYFACES-4431 to jsf-2.3 - NavigationHandlerImpl Class
 MYFACES-4433 to 2.3 - ViewScope Classes
 MYFACES-4394 to jsf-2.3 - JsfArtifactProducer & ImplicitObjectResolver
 MYFACES-4450 to 2.2, 2.3, & 3.0 - CommonPropertyUtils

pnicolucci committed:
Issue #21266: Update MyFaces 2.3 to 2.3.10

volosied commented:
#build

volosied commented on dev/com.ibm.websphere.javaee.jsf.2.3/build.gradle:
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2017, 2021 IBM Corporation and others.
+ * Copyright (c) 2017, 2022 IBM Corporation and others.
```
Update copyrights to 2023? But these weren't touched in 2023, so they might be ok as is?

pnicolucci replied:
If the changes for this were pushed up in 2022 then no need to update the copyright.

volosied committed:
Undo MOJARRA_API_IMP version update

volosied commented on dev/com.ibm.ws.jsfContainer_fat_2.3/fat/src/com/ibm/ws/jsf/container/fat/FATSuite.java:
```diff
@@ -75,9 +75,9 @@ public class FATSuite {
-    public static final String MOJARRA_API_IMP = "publish/files/mojarra/javax.faces-2.3.9.jar";
-    public static final String MYFACES_API = "publish/files/myfaces/myfaces-api-2.3.9.jar";
-    public static final String MYFACES_IMP = "publish/files/myfaces/myfaces-impl-2.3.9.jar";
+    public static final String MOJARRA_API_IMP = "publish/files/mojarra/javax.faces-2.3.10.jar";
```
This should be under another issue, as we're only updating MyFaces, not Mojarra.   We can create a separate issue to update the Mojarra jars.

volosied replied:
Should we update? Mojarra's 2.3.10 release is not necessarily aligned to MyFaces'.

pnicolucci replied:
I'm ok with not updating Mojarra here.

volosied commented:
#build #spawn.fullfat.buckets=com.ibm.ws.jsf.2.3_fat,com.ibm.ws.jsf.2.2_fat

pnicolucci commented on dev/io.openliberty.faces.internal/bnd.bnd:
```diff
@@ -16,6 +16,8 @@
 
 bVersion=1.0
 
+Implementation-Version: 2.3.10
```
This doesn't look to be used?

volosied replied:
Ah, interesting. 

This change was from your commit: https://github.com/pnicolucci/open-liberty/commit/a8c0e79cc18096868a70f4523b605144951e6a94#diff-fe0c218a2f00808c842c8190a1f5dd400113f8a5e925ff282c4678c7c81c3d99R17

But this bnd did use `Implementation-Version` previously.  Now we have `com.ibm.ws.org.apache.myfaces.2.3;version=latest,\` instead.

volosied replied:
Done under: https://github.com/OpenLiberty/open-liberty/commit/4b6d24b3c22518215bb431376f1668847d113b36

volosied replied:
I'll remove this line

pnicolucci reviewed (commented):
I see you removed some of the source classes in the 2.3 impl project but not all of the o.a.m classes. Can you explain? Are these changes not in 2.3.10?

pnicolucci commented on dev/com.ibm.ws.org.apache.myfaces.2.3/bnd.bnd:
```diff
@@ -58,7 +58,7 @@ instrument.classesExcludes: \
 
 Include-Resource: \
-  @${repo;org.apache.myfaces.core:myfaces-impl;2.3.9;EXACT}!/META-INF/**, \
+  @${repo;org.apache.myfaces.core.impl;2.3.10;EXACT}!/META-INF/**, \
```
Why the change from `org.apache.myfaces.core:myfaces-impl` to `org.apache.myfaces.core.impl`?

volosied replied:
I should revert this as it was a change by Jared: https://github.com/OpenLiberty/open-liberty/commit/a732411f4f4d89464fd4f38a47b317810b29604c#diff-8f1167cf80530cee0c71b8dd6375ad80afe10ecdb3cc197b7b0df74847b33ba9R64

Let me make an update. 
I rebased this branch a while ago, but I don't recall getting conflicts.

volosied committed:
Sync latest changes from integration

Remove overlays except MYFACES-4450 (planned for 2.3.11)

Use version variable in bnd

Update bundle version

Issue #21266: remove overlays

volosied commented:
I removed all overlays except for CommonPropertyUtils -- you can se the diff between the overlay and the 2.3.10 source, which is just [MYFACES-4450](https://github.com/apache/myfaces/pull/300/files#diff-6196fe830f0d8252029e57d9fffd119babf135f6d976cc7f17cbac15da785968R534): 
```
/libertyGit/open-liberty/dev/com.ibm.ws.org.apache.myfaces.2.3/src/org/apache/myfaces/shared/renderkit/html/CommonPropertyUtils.java’
534,538c534
<         if ((commonPropertiesMarked & CommonPropertyConstants.ACCESSKEY_PROP) != 0)
<         {
<             HtmlRendererUtils.renderHTMLStringAttribute(writer, component,
<                     HTML.ACCESSKEY_ATTR, HTML.ACCESSKEY_ATTR);
<         }
—
>         renderAccesskeyTabindexProperties(writer, commonPropertiesMarked, component); 
```

Other overlays were removed as there was not significant code difference: 
```

diff ./org/apache/myfaces/cdi/JsfArtifactProducer.java  ‘/Users/volosied/libertyGit/open-
liberty/dev/com.ibm.ws.org.apache.myfaces.2.3/src/org/apache/myfaces/cdi/JsfArtifactProducer.java’
220,221c220
< 
<     /*
—
>        /*
225d223
< 
227d224
< 
244d240
<    


diff ./org/apache/myfaces/el/unified/resolver/implicitobject/ImplicitObjectResolver.java 
‘/Users/volosied/libertyGit/open-liberty/dev/com.ibm.ws.org.apache.myfaces.2.3/src/org/apache/myfaces/el/unified/
resolver/implicitobject/ImplicitObjectResolver.java’
120c120
< 
—
>  
```

volosied commented:
#build #spawn.fullfat.buckets=com.ibm.ws.jsf.2.3_fat,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf_fat_lWAS_jsf2.3

volosied commented on dev/com.ibm.ws.org.apache.myfaces.2.3/bnd.bnd:
```diff
@@ -58,7 +58,7 @@ instrument.classesExcludes: \
 
 Include-Resource: \
-  @${repo;org.apache.myfaces.core:myfaces-impl;2.3.9;EXACT}!/META-INF/**, \
+  @${repo;org.apache.myfaces.core:myfaces-impl;2.3.10;EXACT}!/META-INF/**, \
```
Use {Implementation-Version} instead.

volosied commented:
#build #spawn.fullfat.buckets=com.ibm.ws.jsf.2.3_fat,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf_fat_lWAS_jsf2.3

volosied commented:
#libby

volosied commented:
#build #spawn.fullfat.buckets=com.ibm.ws.jsf.2.3_fat,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf_fat_lWAS_jsf2.3

volosied commented:
#libby
